doc/oper.xml

```diff
@@ -1964,3 +1964,157 @@ gap> ModularProduct(NullDigraph(0), CompleteDigraph(10));
+          is stored</Item>
+    <Item><E>postorder</E>: at each index, the postorder number (order in which the vertex is backtracked on)
+          is stored</Item>
+  </List>
+
+  The record also stores a further 4 attributes.
+  <List>
+    <Item><E>current</E>: the current vertex that is being visited</Item>
+    <Item><E>child</E>: the child of the current vertex</Item>
+    <Item><E>graph</E>: the <A>digraph</A></Item>
+    <Item><E>stop</E>: whether to stop the depth first search</Item>
+  </List>
+  
+  Initially, the current and child attributes will have -1 values and the lists (parent,
+  preorder and postorder) will have -1 values at all of their indicies as no vertex has 
+  been visited. The stop attribute will initially be false.
+  <E>This record should be passed into the ExecuteDFS function.</E>
+  See <Ref Func="ExecuteDFS" Label="for a record, object, integer and 4 functions"/>.
+  <Example><![CDATA[
+gap> record := NewDFSRecord(CompleteDigraph(2));
+rec( child := -1, current := -1, 
+  graph := <immutable complete digraph with 2 vertices>, 
+  parent := [ -1, -1 ], postorder := [ -1, -1 ], 
+  preorder := [ -1, -1 ], stop := false )
+gap> record.preorder;
+[ -1, -1 ]
+gap> record.postorder;
+[ -1, -1 ]
+gap> record.stop;
+false
+gap> record.parent;
+[ -1, -1 ]
+gap> record.child;
+-1
+gap> record.current;
+-1
+gap> record.graph;
+<immutable complete digraph with 2 vertices>
+]]></Example>
+</Description>
+</ManSection>
+<#/GAPDoc>
+
+<#GAPDoc Label="DFSDefault">
+<ManSection>
+  <Oper Name="DFSDefault" Arg="record, data"/>
+  <Description>
+  This is a default function to be passed into the ExecuteDFS function.
+  This does nothing and can be used in place of the PreOrderFunc, PostOrderFunc,
+  AncestorFunc and/or CrossFunc of the ExecuteDFS function.
+  See <Ref Func="ExecuteDFS" Label="for a record, object, integer and 4 functions"/>.
+  <Example><![CDATA[
+gap> PreOrderFunc := function(record, data)
+>       data.num_vertices := data.num_vertices + 1;
+>    end;;
+gap> record := NewDFSRecord(CompleteDigraph(2));;
+gap> data := rec(num_vertices := 0);;
+gap> ExecuteDFS(record, data, 1, PreOrderFunc,
+>               DFSDefault, DFSDefault, DFSDefault);
+gap> data;
+rec( num_vertices := 2 )
+gap> record := NewDFSRecord(CompleteDigraph(2));;
+gap> ExecuteDFS(record, [], 1, DFSDefault,
+>               DFSDefault, DFSDefault, DFSDefault);
+gap> record;
+rec( child := 1, current := 1, 
+  graph := <immutable complete digraph with 2 vertices>, 
+  parent := [ 1, 1 ], postorder := [ 2, 1 ], preorder := [ 1, 2 ], 
+  stop := false )
+]]></Example>
+</Description>
+</ManSection>
+<#/GAPDoc>
+
+<#GAPDoc Label="ExecuteDFS">
+<ManSection>
+  <Oper Name="ExecuteDFS" Arg="record, data, start, PreOrderFunc, PostOrderFunc, AncestorFunc,
+                               CrossFunc"/>
+  <Description>
+  This performs a full depth first search from the <A>start</A> vertex (where <A>start</A> is within the graph).
```

Comment:
"is within" in the sense that it is a vertex of the graph? :)

Reply:
exactly, perhaps "where start is a vertex in the graph" would be more clear?